lib/cli/ui/prompt.rb

```diff
@@ -48,26 +49,29 @@ class << self
         # Free form question when the answer can be empty
         #   CLI::UI::Prompt.ask('What is your opinion on this question?', allow_empty: true)
         #
-        # Question with answers
-        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python))
+        # Multiple choice question with defined handlers
+        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python)) do |handler|
+        #     handler.add_option('rails')  { |selection| puts selection } => outputs "rails" if selected
+        #     handler.add_option('go')     { |selection| puts selection } => outputs "go" if selected
+        #     handler.add_option('ruby')   { |selection| puts selection } => outputs "ruby" if selected
+        #     handler.add_option('python') { |selection| puts selection } => outputs "python" if selected
         #
         #
-        def ask(question, options: nil, default: nil, is_file: nil, allow_empty: true)
-          if (default && !allow_empty) || (options && (default || is_file))
+        def ask(question, default: nil, is_file: nil, allow_empty: true, &define_options)
```

Comment:
I think leave out the `&define_options`, it's inferred and people generally expect it to be referred to by name if it's listed in the signature.